src/core/os_glfw.h

```diff
@@ -305,6 +307,11 @@ void os_on_quit(fn_quit* callback) {
 
 void os_on_focus(fn_focus* callback) {
   glfwState.onWindowFocus = callback;
+  if (callback && glfwState.window) {
+    // make sure whoever sets this callback also gets its initial state
+    int focused = glfwGetWindowAttrib(glfwState.window, GLFW_FOCUSED);
+    callback(focused);
+  }
```

Comment:
The changes here seem really surprising to me.  I don't expect registering an event handler to also fire the event.  What's the motivation here?  As a user of the VR simulator, you'd like `lovr.focus` to get fired at startup?  That is a reasonable request.  Could it be accomplished by adding `os_window_is_focused` and having `desktop_init` do `onFocus(os_window_is_focused());`?

Reply:
Since there is no getter for whether the app is focused, I'm firing the handler so so that the game always knows what the focus state is. In Alloverse, I'm capping the framerate when backgrounded; but without this event, if you defocus before the app fully launches, it won't be capped properly. 

Maybe a better solution is to include an actual getter, e g lovr.system.isFocused(), and lovr.focus event is just used to tell you that that bool has changed. What do you think?

Reply:
- To clarify something, `lovr.focus` does not have anything to do with the window focus.  It's the focus of the headset session.  If the headset module is disabled, `lovr.focus` should never get called.  However, the simulator does use window focus to emulate headset session focus.
- A `lovr.headset.isFocused` would be a welcome change, but might take a lot of work to wrangle all the drivers.
- An `os_window_is_focused` that does `glfwGetWindowAttrib` would be another good change and may be useful for pieces of this.
- A `lovr.system.isWindowFocused` thing would also be okay, but it sort of opens the floodgates and creates an expectation of exposing the entire window API in `lovr.system`, adding all of the necessary events, etc.  Might be good to hold off.
- A small change that seems like it's in the spirit of this one would be to try to get headset drivers to emit focus events when they're initialized (this was the idea I posted above).  If `lovr.headset.isFocused` is added then this can be handled in one place at a higher level, but in the meantime some of the headset drivers could start to do it.

Reply:
I see. Okay, I'll pull this change out for now and we can discuss alternatives separately. Perhaps I should just use https://github.com/Papaew/lovr-window .